File: taotao-parent/taotao-manager-web/src/main/java/com/taotao/manager/controller/BrandController.java
package com.taotao.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.taotao.pojo.TbBrand;
import com.taotao.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * 品牌controller
 */
@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    /**
     * 删除品牌
     */
    @RequestMapping("/delete")
    public Result delete(Long[] ids){
        try{
            brandService.delete(ids);
            return new Result(true,"删除成功");
        }catch(Exception e){
            return new Result(false, "删除失败");
        }
    }

    /**
     * 根据品牌ID获取实体
     */
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }

    /**
     * 修改品牌
     */
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try{
            brandService.update(brand);
            return new Result(true, "修改成功");
        }catch(Exception e){
            return new Result(false, "修改失败");
        }
    }

    /**
     * 新增品牌
     */
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand) {
        try {
            brandService.add(brand);
            return new Result(true, "增加成功");
        } catch (Exception e) {
            return new Result(false, "增加失败");
        }
    }

    /**
     * 条件分页列表
     */
    @RequestMapping("/search")
    public PageResult findByPage(int page, int rows, @RequestBody TbBrand brand){
        return brandService.findPage(page, rows, brand);
    }

    /**
     *返回分页列表
     */
    @RequestMapping("/findPage")
    public PageResult findByPage(int page, int rows){
       return brandService.findPage(page,rows);
    }

    /**
     * 返回全部品牌列表
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }

    /**
     * 品牌下拉框数据
     */
    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }
}

File: taotao-parent/taotao-manager-web/src/main/webapp/js/controller/brandController.js
//品牌控制层
app.controller('brandController',function($scope,$controller,brandService){

    $controller('baseController',{$scope:$scope});//继承

    //查询所有品牌
    $scope.findAll = function () {
        brandService.findAll().success(
            function (response) {
                $scope.list = response;
            }
        )
    };

    //分页查询
    $scope.findPage = function(page, rows){
        brandService.findPage(page,rows).success(
            function (response) {
                $scope.list = response.rows;//获取当前分页列表
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }

        )
    };

    //定义查询的条件对象
    $scope.searchEntity = {};
    //条件分页查询
    $scope.search = function (page,rows) {
        brandService.search(page,rows,$scope.searchEntity).success(
            function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems = response.total;
            }

        )
    };

    //新增品牌
    $scope.save = function () {
        if($scope.entity.id != null){
            brandService.update($scope.entity).success(
                function (response) {
                    if(response.success){
                        $scope.reloadList();//重新查询
                    }else{
                        alert(response.message);
                    }
                }
            )
        }else {
            brandService.add($scope.entity).success(
                function (response) {
                    if(response.success){
                        $scope.reloadList();//重新查询
                    }else{
                        alert(response.message);
                    }
                }
            )
        }


    };

    //根据品牌ID进行回显
    $scope.findOne = function (id) {
        brandService.findOne(id).success(
            function (response) {
                $scope.entity = response;
            }
        )
    };

    //批量删除品牌列表
    $scope.delete = function () {
        brandService.delete($scope.selectIds).success(
            function (response) {
                if(response.success){
                    $scope.reloadList();
                }else{
                    alert(response.message);
                }
            }
        )
    };
});